### PR TITLE
Added metric prefix macros

### PR DESCRIPTION
**Pull request by larsewi**
I see extensively in the source code that sizes are calculated like this `5 * 1024 * 1024`. I believe it would be more readable if we used metric prefix macros like `MEBIBYTE(5)`.

### CODE REVIEW

**Comment by larsewi** [comment]
I should probably add unit tests I guess.

File: libutils/definitions.h [hunk 1]
```diff
@@ -25,6 +25,14 @@
 #ifndef CFENGINE_DEFINITIONS_H
 #define CFENGINE_DEFINITIONS_H
 
+/*****************************************************************************
+ * Metric prefixes
+ *****************************************************************************/
+#define KIBIBYTE(n) n * 1024
+#define MEBIBYTE(n) n * 1024 * 1024
+#define GIBIBYTE(n) n * 1024 * 1024 * 1024
+#define TEBIBYTE(n) n * 1024 * 1024 * 1024 * 1024
```
Comment (olehermanse) [inline libutils/definitions.h:34]: I would put them in parenthesis, although it probably doesn't matter in most cases.

```suggestion
#define KIBIBYTE(n) (n * 1024)
#define MEBIBYTE(n) (n * 1024 * 1024)
#define GIBIBYTE(n) (n * 1024 * 1024 * 1024)
#define TEBIBYTE(n) (n * 1024 * 1024 * 1024 * 1024)
```

File: tests/unit/definitions_test.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    assert_int_equal(GIBIBYTE(2), 2147484000);
+    assert_int_equal(GIBIBYTE(6), 6442451000);
+    assert_int_equal(GIBIBYTE(10), 10737420000);
+    assert_int_equal(GIBIBYTE(18), 19327350000);
+
+    assert_int_equal(TEBIBYTE(3), 3298535000000);
+    assert_int_equal(TEBIBYTE(7), 7696581000000);
+    assert_int_equal(TEBIBYTE(11), 12094630000000);
+    assert_int_equal(TEBIBYTE(19), 20890720000000);
```
Comment (olehermanse) [inline tests/unit/definitions_test.c:26]: Adding some random numbers is fine, but I would test the basic 0, 1, and 2 for each of them, for example:

```
    assert_int_equal(GIBIBYTE(0), 0);
    assert_int_equal(GIBIBYTE(1), 1024 * MEBIBYTE(1));
    assert_int_equal(2 * GIBIBYTE(1), GIBIBYTE(2));
    assert_int_equal(GIBIBYTE(6), 6442451000);
    assert_int_equal(GIBIBYTE(10), 10737420000);
    assert_int_equal(GIBIBYTE(18), 19327350000);
```

**Review by olehermanse — CHANGES_REQUESTED** [review]
LGTM, but there seems to be some problem with your last commit. Please make sure you are committing with an e-mail address that you have added to your GitHub account.

File: tests/unit/definitions_test.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+int main()
+{
+    PRINT_TEST_BANNER();
+    const UnitTest tests[] =
+    {
+        unit_test(test_metric_prefixes())
```
Comment (olehermanse) [inline tests/unit/definitions_test.c:34]:
```suggestion
        unit_test(test_metric_prefixes)
```

Give it the function pointer, don't call the function.

File: libutils/definitions.h [hunk 4]
```diff
@@ -25,6 +25,14 @@
 #ifndef CFENGINE_DEFINITIONS_H
 #define CFENGINE_DEFINITIONS_H
 
+/*****************************************************************************
+ * Metric prefixes
+ *****************************************************************************/
+#define KIBIBYTE(n) (n * 1024)
+#define MEBIBYTE(n) (n * 1024 * 1024)
+#define GIBIBYTE(n) (n * 1024 * 1024 * 1024)
+#define TEBIBYTE(n) (n * 1024 * 1024 * 1024 * 1024)
```
Comment (olehermanse) [inline libutils/definitions.h:34]:
```suggestion
#define KIBIBYTE(n) (n * 1024L)
#define MEBIBYTE(n) (n * 1024L * 1024L)
#define GIBIBYTE(n) (n * 1024L * 1024L * 1024L)
#define TEBIBYTE(n) (n * 1024L * 1024L * 1024L * 1024L)
```

I guess you have to write it like this to avoide overflow.

**Review by olehermanse — CHANGES_REQUESTED** [review]
Some more things need fixing.

**Review by vpodzime — APPROVED** [review]
Looks good to me.

**Review by olehermanse — CHANGES_REQUESTED** [review]
Please fix the commit so that GH recognizes who wrote this.

I guess you have to fix your git config, then do `git commit --amend --reset-author`.

**Review by vpodzime — APPROVED** [review]
ACK

**Review by olehermanse — APPROVED** [review]

**Pull request merged**